src/app-data/service-api.ts

```diff
@@ -33,6 +40,69 @@ const plugin: FastifyPluginAsync = async (fastify) => {
 
     fastify.addHook('preHandler', fastify.verifyBearerAuth);
 
+    // Files
+    const FILE_ITEM_TYPES = {
+      S3: 's3File',
+      LOCAL: 'file',
+    };
+    const SERVICE_ITEM_TYPE = enableS3 ? FILE_ITEM_TYPES.S3 : FILE_ITEM_TYPES.LOCAL;
+
+    const randomHexOf4 = () => ((Math.random() * (1 << 16)) | 0).toString(16).padStart(4, '0');
+    const buildFilePath = () => {
+      const filepath = `${randomHexOf4()}/${randomHexOf4()}/${randomHexOf4()}-${Date.now()}`;
+      return `apps/${filepath}`;
+    };
+
+    fastify.register(GraaspFilePlugin, {
+      serviceMethod: enableS3 ? ServiceMethod.S3 : ServiceMethod.LOCAL,
+      serviceOptions: {
+        s3: fastify.s3FileItemPluginOptions,
+        local: fastify.fileItemPluginOptions,
+      },
+      buildFilePath: buildFilePath,
+
+      uploadPreHookTasks: async ({ parentId: itemId }, { token }) => {
+        const { member: id } = token;
+        return [ taskManager.createGetTask({ id }, itemId, { visibility: 'member'}, token) ];
+      },
+      uploadPostHookTasks: async (
+        { filename, itemId, filepath, size, mimetype },
+        { token }
+      ) => {
+        const { member: id } = token;
+
+        const name = filename.substring(0, 100);
```

Comment:
I think this `100` comes from some strip from file-item? Or maybe factor out the number at least.